grakn-graph/src/test/java/ai/grakn/generator/MetaTypeNames.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.generator;
+
+import ai.grakn.concept.TypeName;
+import ai.grakn.util.Schema;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.stream.Stream;
+
+import static ai.grakn.graql.internal.util.CommonUtil.toImmutableSet;
+
+/**
+ * Generator that generates meta type names only
+ */
+public class MetaTypeNames extends AbstractGenerator<TypeName> {
```

Comment:
It's used in just one test: `whenCallingClear_AllMetaConceptsArePresent`. If I remove that call we could merge this into the `TypeNames` generator. It's still helpful for this to exist (but maybe not in its own class), similar to `bar` above: it catches edge cases when using meta type names where they are not expected.